plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -257,4 +264,25 @@ protected static List<Column> toHiveColumns(List<NestedField> columns)
                         Optional.empty()))
                 .collect(toImmutableList());
     }
+
+    protected CommitStatus checkCommitStatus(String metadataLocation)
+    {
+        AtomicReference<CommitStatus> status = new AtomicReference<>(CommitStatus.UNKNOWN);
```

Comment:
looks like AtomicReference is used only to determine whether retrying succeeded.
sounds like there should be a simpler way to do that